api/src/pcapi/core/finance/repository.py

```diff
@@ -44,43 +44,90 @@ def get_business_units_query(  # type: ignore [no-untyped-def]
+    )
+    venue_subquery = offerers_models.Venue.query
+    if not user.has_admin_role:
+        venue_subquery = venue_subquery.join(
+            offerers_models.UserOfferer,
+            offerers_models.Venue.managingOffererId == offerers_models.UserOfferer.offererId,
+        ).filter(offerers_models.UserOfferer.user == user, offerers_models.UserOfferer.isValidated)
+    if offerer_id:
+        venue_subquery = venue_subquery.filter(offerers_models.Venue.managingOffererId == offerer_id)
+    if venue_subquery.whereclause is not None:
+        venue_subquery = venue_subquery.with_entities(offerers_models.Venue.id).subquery()
+        query = query.filter(offerers_models.Venue.id.in_(venue_subquery))
+    return query
+
+
 def find_business_unit_by_siret(siret: str) -> models.BusinessUnit | None:
     return models.BusinessUnit.query.filter_by(siret=siret).one_or_none()
 
 
-def get_invoices_query(  # type: ignore [no-untyped-def]
+def get_invoices_query(
     user: users_models.User,
-    business_unit_id: int = None,
-    date_from: datetime.date = None,
-    date_until: datetime.date = None,
-):
+    business_unit_id: int | None = None,
+    reimbursement_point_id: int | None = None,
+    date_from: datetime.date | None = None,
+    date_until: datetime.date | None = None,
+) -> sqla_orm.Query:
     """Return invoices for the requested offerer.
 
     If given, ``date_from`` is **inclusive**, ``date_until`` is
     **exclusive**.
     """
-    business_units_subquery = offerers_models.Venue.query
-    if not user.has_admin_role:
-        business_units_subquery = business_units_subquery.join(
-            offerers_models.UserOfferer,
-            offerers_models.UserOfferer.offererId == offerers_models.Venue.managingOffererId,
-        ).filter(offerers_models.UserOfferer.user == user, offerers_models.UserOfferer.isValidated)
-    if business_unit_id:
-        # Filtering like this makes sure that the requested business
-        # unit id is accessible by the requesting user.
-        business_units_subquery = business_units_subquery.filter(
-            offerers_models.Venue.businessUnitId == business_unit_id
+    if FeatureToggle.ENABLE_NEW_BANK_INFORMATIONS_CREATION.is_active():
+        reimbursement_point_subquery = offerers_models.Venue.query
+
+        if not user.has_admin_role:
+            reimbursement_point_subquery = reimbursement_point_subquery.join(
+                offerers_models.UserOfferer,
+                offerers_models.UserOfferer.offererId == offerers_models.Venue.managingOffererId,
+            ).filter(offerers_models.UserOfferer.user == user, offerers_models.UserOfferer.isValidated)
+
+        if reimbursement_point_id:
+            reimbursement_point_subquery = reimbursement_point_subquery.join(
+                offerers_models.VenueReimbursementPointLink,
+                offerers_models.VenueReimbursementPointLink.venueId == offerers_models.Venue.id,
```

Comment:
(détail / question de goût) Je pense qu'on peut utiliser ici directement le nom de la relation, qui rend le code un peu plus concis :

```suggestion
                offerers_models.Venue.reimbursement_point_links
```